sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/DoFnSchemaInformation.java

```diff
@@ -29,25 +38,175 @@
    * The schema of the @Element parameter. If the Java type does not match the input PCollection but
    * the schemas are compatible, Beam will automatically convert between the Java types.
    */
-  @Nullable
-  public abstract SchemaCoder<?> getElementParameterSchema();
+  public abstract List<SerializableFunction<?, ?>> getElementConverters();
 
   /** Create an instance. */
   public static DoFnSchemaInformation create() {
-    return new AutoValue_DoFnSchemaInformation.Builder().build();
+    return new AutoValue_DoFnSchemaInformation.Builder()
+        .setElementConverters(Collections.emptyList())
+        .build();
   }
 
   /** The builder object. */
   @AutoValue.Builder
   public abstract static class Builder {
-    abstract Builder setElementParameterSchema(@Nullable SchemaCoder<?> schemaCoder);
+    abstract Builder setElementConverters(List<SerializableFunction<?, ?>> converters);
 
     abstract DoFnSchemaInformation build();
   }
 
   public abstract Builder toBuilder();
 
-  public <T> DoFnSchemaInformation withElementParameterSchema(SchemaCoder<T> schemaCoder) {
-    return toBuilder().setElementParameterSchema(schemaCoder).build();
+  DoFnSchemaInformation withElementParameterSchema(
```

Comment:
Would you mind adding a docstring here and on `withUnboxParameter`? I think that would help a lot with readability. It doesn't need to be exhaustive, but there are a couple of questions I had to work through by reading  the code that I think could be clarified here: 
- Can `elementCoder` be considered `outputCoder` (i.e. the coder to use for selectOutputSchema), or is it something else?
- What's the difference between `withElementParameterSchema` with `unbox = true` and `withUnboxParameter`?

Reply:
Done